view/frontend/templates/button_product_page.phtml

```diff
@@ -256,7 +256,21 @@ $trackCallbacks = $block->getTrackCallbacks();
 
                 var buttonClass = "bolt-product-checkout-button";
                 var hints = {prefill:{}};
```

Comment:
@vitaliyreznikov 
Perhaps easier and (to me at least) straightforward way to do this was to just inject hints into the template,
something like:
```javascript
var hints = '<?=$block->getHints();?>';
```
without the need of a controller.

Reply:
@danac-gs Are we speaking about "empty" hints?
So do you propose to create `function getHints() { return "{prefill:{}}"; }` ?
In general, we need to receive Hints from server by Ajax call.

Reply:
@vitaliyreznikov 
No. I mean the complete hints (if there are any). The hints value would be the same at the time the page renders as it would be later, when fetched via Ajax call. No need to receive it from server, can add it to the page when the page is generated on the server. We can only get hints for a logged in customer. No controller, less code to maintain. $block::getHints() that calls \Bolt\Boltpay\Helper\Cart::getHints and formats the data accordingly. Perhaps add another flag to Cart::getHints to prevent it to load the quote. It creates one otherwise ($this->checkoutSession->getQuote()).

Reply:
@danac-gs Correct me if I wrong but product pages are cached, so we can't receive hints when the page renders.

Reply:
@vitaliyreznikov 100% correct! Scratch everything I wrote above except the last two lines. You don't need a flag, you can pass different $checkoutType ('product') and skip fetching the quote.

Reply:
@danac-gs Ok, sounds reasonable. Done.